### ISSUE TITLE
Freature Request: Add an option to search with ripgrep

### ISSUE THREAD

**Issue by bgianfo** [vrothberg/vgrep#60]
ripgrep (https://github.com/BurntSushi/ripgrep) is one of the fastest tools for searching code. It would be nice if vgrep supported ripgrep as a search backend.

**Comment by vrothberg** [comment]
Hi @bgianfo, thanks for reaching out!

If you want to contribute, `vgrep` can easily be extended to support more backends. https://github.com/vrothberg/vgrep/blob/master/vgrep.go#L185 is a good point to add `ripgrep` and run it if it's in `$PATH`. `vgrep` expects the output to be split by null-bytes as `$file\0$line\0$content`.

**Comment by bartol** [comment]
Hi,

I was playing with this feature this afternoon and although it isn't done, I've made some progress. Only issue currently is ripgrep's color output. It escapes all colors or none. Grep escapes only match color.

Output that I am getting:
```
panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
main.(*vgrep).splitMatch(0xc0000242a0, 0xc0000dc000, 0x3a, 0xc0000e0001, 0x0, 0x0, 0xc0000bc380, 0x36, 0x37, 0x0)
	/home/bartol/dev/vgrep/vgrep.go:249 +0x3f8
main.(*vgrep).grep(0xc0000242a0, 0xc00005c540, 0x1, 0x1)
	/home/bartol/dev/vgrep/vgrep.go:233 +0x3c5
main.main()
	/home/bartol/dev/vgrep/vgrep.go:105 +0x470
```
[changes](https://github.com/bartol/vgrep/commit/d854fb10fe6a55ceef42cf1f55ded30293cdd73b)

This is my first time contributing to open source project (you have to start somewhere) and I've never touched Golang code in my life so it isn't perfect, but I've tried my best.

If someone more experienced could guide me to solve issue or complete this feature that would be really awesome!

**Comment by vrothberg** [comment]
Hi @bartol,

thanks for working on that! I suggest to open a pull request, so we can through the CI at it and make reviewing easier. Would you open one?

The parsing code is _very_ specific to grep, so I guess we have to do some massaging to support ripgrep.

**Comment by vrothberg** [comment]
Fixed with #65, thanks everyone!
